lib/js/jsbigints.nim

```diff
@@ -0,0 +1,137 @@
+## Arbitrary precision integers.
+## * https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt
+when not defined(js) and not defined(nimdoc):
+  {.fatal: "Module jsbigints is designed to be used with the JavaScript backend.".}
+
+type JsBigInt* = ref object of JsRoot ## Arbitrary precision integer for JavaScript target.
+
+func newBigInt*(integer: SomeInteger): JsBigInt {.importjs: "BigInt(#)".}
+  ## Constructor for `JsBigInt`.
+
+func big*(integer: cstring): JsBigInt {.importjs: "BigInt(#)".} =
+  ## Constructor for `JsBigInt`.
+  runnableExamples:
+    doAssert big"-1" == big"1" - big"2"
+
+func toLocaleString*(this: JsBigInt): cstring {.importjs: "#.$1()".}
+  ## https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt/toLocaleString
+
+func toLocaleString*(this: JsBigInt; locales: cstring): cstring {.importjs: "#.$1(#)".}
+  ## https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt/toLocaleString
+
+func toLocaleString*(this: JsBigInt; locales: openArray[cstring]): cstring {.importjs: "#.$1(#)".}
+  ## https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt/toLocaleString
+
+func toString*(this: JsBigInt; radix: int): cstring {.importjs: "#.$1(#)".}
+  ## https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt/toString
+
+func toString*(this: JsBigInt): cstring {.importjs: "#.toString()".}
+  ## https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt/toString
+
+func `$`*(this: JsBigInt): string = $toString(this)
+  ## Return a string representation of `JsBigInt`.
+
+func asIntN*(width: int; bigInteger: JsBigInt): int {.importjs: "BigInt.$1(#, #)".}
+  ## https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt/asIntN
+
+func asUintN*(width: int; bigInteger: JsBigInt): int {.importjs: "BigInt.$1(#, #)".}
+  ## https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt/asUintN
+
+func `+`*(x, y: JsBigInt): JsBigInt {.importjs: "(# $1 #)".}
+
+func `-`*(x, y: JsBigInt): JsBigInt {.importjs: "(# $1 #)".}
+
+func `*`*(x, y: JsBigInt): JsBigInt {.importjs: "(# $1 #)".}
+
+func `div`*(x, y: JsBigInt): JsBigInt {.importjs: "(# / #)".}
+   ## Same as `div` but for `JsBigInt`.
+
+func `mod`*(x, y: JsBigInt): JsBigInt {.importjs: "(# % #)".}
+  ## Same as `mod` but for `JsBigInt` (uses JavaScript `BigInt() % BigInt()`).
+
+func `+=`*(x, y: JsBigInt) {.importjs: "(# $1 #)".}
+
+func `-=`*(x, y: JsBigInt) {.importjs: "(# $1 #)".}
+
+func `*=`*(x, y: JsBigInt) {.importjs: "(# $1 #)".}
+
+func `div=`*(x, y: JsBigInt) {.importjs: "(# /= #)".}
+
+func `mod=`*(x, y: JsBigInt) {.importjs: "(# $1 #)".}
```

Comment:
`importjs: "(# $1 #)"` can't work here, there are 2 options:

```nim
func `mod=`*(x, y: JsBigInt) {.importjs: "(# %= #)".}
`mod=`(a, big"3") # ugly but matches `mod`

func `%=`*(x, y: JsBigInt) {.importjs: "(# $1 #)".}
a %= big"3" # nice but doesn't match `mod`
```

what do you guys think?

Reply:
I can not make `mod=` work, it complains, thinks it is an assign or a mod or a combination of both. I used `%=`.